lib/helium/collection.rb

```diff
@@ -0,0 +1,129 @@
+module Helium
+  class Collection
+    include Enumerable
+
+    attr_reader :filter_criteria
+
+    def initialize(opts)
+      @client     = opts.fetch(:client)
+      @klass      = opts.fetch(:klass)
+      @belongs_to = opts.fetch(:belongs_to, nil)
+
+      @filter_criteria = {}
+    end
+
+    # Returns all resources
+    # @option opts [Client] :client A Helium::Client
+    # @return [Helium::Collection] a Collection of all of the given Resource
+    def all
+      @filter_criteria = {}
+      self
+    end
+
+    # Uses metadata filtering
+    # (https://docs.helium.com/api/v1/metadata/index.html#filtering)
+    # to search for a collection of resources matching the search parameters
+    # @param [Hash] a set of search criteria
+    #
+    # @example Search for sensors by location
+    #   client.sensors.where(location: 'Building B') #=> [Sensor, Sensor]
+    #
+    # @example Search for multiple matching search parameters
+    #   client.sensors.where(departments: ['it', 'engineering']) #=> [Sensor, Sensor]
+    #
+    # @return [Collection] a Collection of resources matching the provided search criteria
+    def where(criteria)
+      add_filter_criteria(criteria)
+      self
+    end
+
+    # Returns an array of the Resources belonging to the Collection
+    # @return [Resource]
+    def collection
+      fetch_collection
+    end
+
+    def each
```

Comment:
I figured as much b/c the rest of enumerable just worked.  I just couldn't find an interface definition like you'd see in java/c++ that states the bare minimum you need to overload the operators.